src/Functions/like.cpp

```diff
@@ -11,11 +11,28 @@ struct NameLike
     static constexpr auto name = "like";
 };
 
-using FunctionLike = FunctionsStringSearch<MatchImpl<true>, NameLike>;
+struct NameILike
+{
+    static constexpr auto name = "ilike";
+};
+
+namespace
+{
+    using LikeImpl = MatchImpl</*SQL LIKE */ true, /*revert*/false>;
+    using ILikeImpl = MatchImpl<true, false, /*case-insensitive*/true>;
+}
+
+using FunctionLike = FunctionsStringSearch<LikeImpl, NameLike>;
+using FunctionIlike = FunctionsStringSearch<ILikeImpl, NameILike>;
 
 void registerFunctionLike(FunctionFactory & factory)
 {
     factory.registerFunction<FunctionLike>();
 }
 
+void registerFunctionIlike(FunctionFactory & factory)
```

Comment:
I prefer creating a separate file `ilike.cpp`
BTW, do we have NOT ILIKE operator?

Reply:
> I prefer creating a separate file `ilike.cpp`

ILIKE and LIKE differ only in the searcher (VolnitskyCaseInsensitive vs. Volnitsky), so I don't think this is necessary (I wouldn't rather pollute the file structure).
Same is true for NOT LIKE and NOT ILIKE.

> BTW, do we have NOT ILIKE operator?

Just discussed with my colleague!
Not yet, but I can add it.